Makefile

```diff
@@ -45,13 +58,29 @@ vet:
 # build targets
 #
 
-bin/%:
-	$(Q)echo "Building $@..."; \
-	$(GO_BUILD) -o $@ ./$(subst bin/,cmd/,$@)
+$(BINARIES): bin/%:
+	$(Q)echo "Building $@..."
+	$(Q)(cd cmd/$(*) && $(GO_BUILD) -o $(abspath $@) .)
+
+#
+# vendoring targets
+#
+.PHONY: mod-verify mod-tidy $(MOD_TIDY_TARGETS) $(MOD_VERIFY_TARGETS)
+
+MOD_TARGETS := $(CMDS) ..
+MOD_TIDY_TARGETS := $(patsubst %,mod-tidy-%,$(MOD_TARGETS))
+MOD_VERIFY_TARGETS := $(patsubst %,mod-verify-%,$(MOD_TARGETS))
+
+$(MOD_TIDY_TARGETS): mod-tidy-%:
```

Comment:
What is the purpose of `mod-tidy-%` here ? AFAICT, it will evaluate to empty.

Reply:
This is how pattern targets are constructed. The `%` in the rule matches the last part of the target and allows `$(*)` to be used in the commands.

Reply:
Have you tried checking what, for instance, that `mod-tidy-%` evaluates to in the `$(MOD_TIDY_TARGETS): mod-tidy-%` rule with something like this:
```
$(MOD_TIDY_TARGETS): mod-tidy-%:
	echo "*** prerequisites for '$@' => '$^' ***"
	$(Q)echo "Running $@... in $(abspath ./cmd/$(*))"
	$(Q)(cd $(abspath ./cmd/$(*)) && ${GO} mod tidy)
```

Reply:
That's not how pattern targets are constructed IMO. In a pattern target there must be a  `%` (exactly one) in the target. Then you can have one or more `%` in the prerequisites which are then replaced with the same stem. In $(MOD_TIDY_TARGETS) there are 0 `%`s. That's where my whole confusion is coming from. The way how I originally misread you rule definition definitely does not make any sense. The way how I now more carefully read your rule... I don't understand what it means.

Reply:
In the rule as written:
```
$(MOD_TIDY_TARGETS): mod-tidy-%: 
```
there are **no** prerequisites defined. This is a [static pattern rule](https://web.mit.edu/gnu/doc/html/make_4.html#SEC37) of the form:
```
targets ...: target-pattern: dep-patterns ...
        commands
```
In this case, `dep-patterns` is empty.

From the docs:
> The target-pattern and dep-patterns say how to compute the dependencies of each target. Each target is matched against the target-pattern to extract a part of the target name, called the stem. This stem is substituted into each of the dep-patterns to make the dependency names (one from each dep-pattern).

If we then expand this we have:
```
mod-tidy-cdi mod-tidy-validate: mod-tidy-%:
```
Meaning that this will effectively result in two "rules" with the *stem* matching `cdi` and `validate` respectively. This ensures that the commands associated with the rule have `$(*)` set to `cdi` and `validate` repspectively.

As a note, I use these kinds of rules in our Makefiles to cover cases where we are adding new commands, for example. Here, I generate the top-level targets using wildcards and use replacements and static pattern rules to define generic targets. This means that no make changes are required to add new commands.

If we think that this is excessive for this repo, I can expand these rules instead.

Reply:
> there are **no** prerequisites defined. This is a [static pattern rule](https://web.mit.edu/gnu/doc/html/make_4.html#SEC37) of the form:

Thank you ! That is exactly the snippet I was looking for in the same piece of documentation yesterday, but I failed to find it.